src/lib/reasoners/adt_rel.ml

```diff
@@ -55,17 +55,83 @@ type t =
        inconsistency. *)
 
     domains : (HSS.t * Explanation.t) MX.t;
+    (* Map of the uninterpreted semantic values to domains of their possible
+       constructors. The explanation justifies that any value assigned to
+       the semantic value has to use a constructor lying in the domain. *)
+
+    (* TODO: rename this field. *)
     seen_destr : SE.t;
+    (* Set of all the guarded destructors known by the theory. A guarded
+       destructor isn't interpreted by the ADT theory.
+
+       This field is used to prevent transforming twice a guarded
+       destructor to its non-guarded version and for debugging purposes. *)
+
+    (* TODO: rename this field. *)
     seen_access : SE.t;
+    (* Set of all the non-guarded destructors known by the theory. Contrary
+       to a guarded destructor, a non-guarded destructor is interpreted
+       by the ADT theory. *)
+
     seen_testers : HSS.t MX.t;
     [@ocaml.ppwarning "selectors should be improved. only representatives \
                        in it. No true or false _is"]
+    (* TODO: This code is outdated. It seems that the theory doesn't need
+       to register all the testers in [add_aux]. Now this field is only
+       read to remember we have already forced a constructor because we have
+       assumed its associated tester. *)
 
     selectors : (E.t * Ex.t) list MHs.t MX.t;
+    (* Map of pending destructor equations. A destructor equation on an
+       ADT term [t] is an equation of the form:
+         d t = d' t
+       where [d] is a guarded destructor and [d'] its non-guarded version.
+
+       More precisely, this map matches a representative [r] with
+       all the destructor equations on terms of the class of [r]. These
+       destructor equations are classified by their corresponding
+       constructor.
+
+       Consider [d] a guarded destructor and [c] its associated constructor.
+
+       - When we register the destructor application [d t] in [add_aux], we
+         produce an equation [d t = d' t] where [d'] is the non-guarded
+         version of [d]. If we don't already know that the value of [t] has
+         to use the constructor [c], we put the equation [d t = d' t] in this
+         map. Otherwise we propagate the equation to CC(X).
```

Comment:
Can you clarify the mapping? I assume it is `d t -> c -> [d t = d' t, ...]`?

Reply:
I guess my comment is unclear but the mapping is not as you think. 
For some representative (semantic value) `r`, the map contains a map of constructors of `X.type_info r` to destructor equations `d t = d' t` where `t` is a term of the class of `r`.